### DISCUSSION TITLE
Too high loss while training NER with custom entities

### DISCUSSION THREAD

**Discussion by AlexFridman** [explosion/spaCy#7086] [Help: Coding & Implementations]
Hello! I faced with too high losses training custom NER (2000+). 
I use the pipe to train with a default config (en large)

```
python -m spacy train config.cfg --output ./output --paths.train ./train.spacy --paths.dev ./dev.spacy
```
From other discussions I see -- the problem is because my data does not contain predefined entities. Is it possible to completely reset the list of entities I train in the config file?

```
E    #       LOSS TOK2VEC  LOSS NER  ENTS_F  ENTS_P  ENTS_R  SCORE 
---  ------  ------------  --------  ------  ------  ------  ------
  0       0          0.00     40.00    9.57    6.09   22.33    0.10
  0     200         35.70   2163.45   83.16   83.79   82.53    0.83
  0     400         11.09   1182.04   88.91   89.03   88.78    0.89
  0     600         13.27   1150.09   89.98   90.20   89.77    0.90
  0     800         19.24   1087.27   92.02   91.73   92.32    0.92
  0    1000         25.08   1343.08   93.06   92.75   93.38    0.93
```

Could you also point me to the NER format description. I understand what is 
B-..
I-..,
but confused with L-, U-.

Currently, my dataset built with B- and I- tags.

Thank you for your help!

Best Wishes,
Alex

- **spaCy version:** 3.0.1
- **Platform:** Linux-5.8.0-43-generic-x86_64-with-glibc2.29
- **Python version:** 3.8.3
- **Pipelines:** en_core_web_trf (3.0.0), en_core_web_lg (3.0.0)

**Comment by svlandeg** [comment]
Hi! I'll be moving this to the discussion forum that is better suited for discussing best practices and use-case specific help from the community. It may look like this issue got closed/locked, but it's really just moved ;-)

**Comment by svlandeg** [comment]
Hi!

>  I faced with too high losses training custom NER (2000+)

I see what you mean, the training loss is indeed quite high. On the other hand, the F-score on your dev data doesn't seem to be so bad? And in the training log you've shared, it's still going up, so training seems to be working quite well in my opinion?

> Is it possible to completely reset the list of entities I train in the config file?

You can create a new NER component from scratch, and train that, if you don't need the entity types that were available in the pipelines you downloaded. (if you share your config file, it's easier for us to see what exactly you're doing now and how you could change that)

> Could you also point me to the NER format description.

You can find some more information on the BILUO scheme here: https://spacy.io/usage/linguistic-features#accessing-ner
Also have a look at the documentation on the `convert` CLI tool, which includes conversion from a few common NER formats: https://spacy.io/api/cli#converters

Basically what you need for training is the new binary .spacy format, but it looks like you've already got those?

Anyway if you have more questions - shoot ;-)

**Reply by AlexFridman** [reply]
Hello! Thank you for the reply.

Yes, the metrics are quite well. I'm going through the [basic example](https://spacy.io/usage/training#quickstart).
I generated a config file and a binary train & dev data.

Is it possible to "create a new NER component" changing the smth in the config file?

**Reply by svlandeg** [reply]
If you create a new config file with the quickstart, or the `init config` command, it will be a brand new NER model, defined in the section `[components.ner]`. You'll see that it starts with `factory = "ner"`, which is a built-in factory that calls the constructor of the Named Entity Recognizer.

I'm not sure I understand what you want to change specifically?
